### PR TITLE
Abass/card ui

### PR DESCRIPTION
**Pull request by abassawo**
Category card list UI changes

### CODE REVIEW

File: app/src/main/res/layout/category_row_item.xml [hunk 1]
```diff
@@ -1,35 +1,65 @@
...
+        </RelativeLayout>
+    </android.support.v7.widget.CardView>
 
-</android.support.v7.widget.CardView>
+    <ImageView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:background="@android:color/transparent"
+        android:padding="0dp"
+        android:src="@drawable/group_6"
```
Comment (gsintrepid) [inline app/src/main/res/layout/category_row_item.xml]: Can you change the name of this drawable to be more meaningful?

File: app/src/main/res/layout/category_row_item.xml [hunk 2]
```diff
@@ -1,35 +1,65 @@
...
+                />
+        </RelativeLayout>
+    </android.support.v7.widget.CardView>
 
-</android.support.v7.widget.CardView>
+    <ImageView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:background="@android:color/transparent"
+        android:padding="0dp"
```
Comment (gsintrepid) [inline app/src/main/res/layout/category_row_item.xml]: Remove padding

File: app/src/main/res/layout/category_row_item.xml [hunk 3]
```diff
@@ -1,35 +1,65 @@
...
+                android:src="@drawable/ic_menu_gray_24dp"
+                />
+        </RelativeLayout>
+    </android.support.v7.widget.CardView>
 
-</android.support.v7.widget.CardView>
+    <ImageView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:background="@android:color/transparent"
```
Comment (gsintrepid) [inline app/src/main/res/layout/category_row_item.xml]: Is this background making any difference? What's the default?

File: app/src/main/res/layout/category_row_item.xml [hunk 4]
```diff
@@ -1,35 +1,65 @@
...
+                android:layout_height="wrap_content"
+                android:layout_below="@+id/category_item_title"
+                />
 
-    </LinearLayout>
+            <ImageView
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_alignParentEnd="true"
+                android:layout_marginEnd="21dp"
```
Comment (gsintrepid) [inline app/src/main/res/layout/category_row_item.xml]: Make this a dimension variable too

File: app/src/main/res/layout/category_row_item.xml [hunk 5]
```diff
@@ -1,35 +1,65 @@
...
-            android:id="@+id/category_item_description"
-            android:layout_width="match_parent"
-            android:layout_height="wrap_content"
-            />
+            <TextView
+                android:id="@+id/category_item_title"
+                style="@style/AvenirNext_DemiBold"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_marginTop="2dp"
```
Comment (gsintrepid) [inline app/src/main/res/layout/category_row_item.xml]: This is a very small margin, you probably don't need it. If you do, make it a dimension variable

File: app/src/main/res/layout/category_row_item.xml [hunk 6]
```diff
@@ -1,35 +1,65 @@
...
+            >
 
-        <TextView
-            android:id="@+id/category_item_description"
-            android:layout_width="match_parent"
-            android:layout_height="wrap_content"
-            />
+            <TextView
+                android:id="@+id/category_item_title"
+                style="@style/AvenirNext_DemiBold"
```
Comment (gsintrepid) [inline app/src/main/res/layout/category_row_item.xml]: Use `android:fontFamily` instead of `style`

File: app/src/main/res/layout/category_row_item.xml [hunk 7]
```diff
@@ -1,35 +1,65 @@
...
+        android:id="@+id/scoring_category_card_view"
+        style="@style/AppTheme.IconButtonCardView"
         android:layout_width="match_parent"
-        android:layout_height="match_parent"
-        android:orientation="vertical"
+        android:layout_height="@dimen/category_card_height"
+        android:layout_gravity="center"
+        android:layout_marginBottom="@dimen/activity_vertical_margin"
+        android:layout_marginTop="@dimen/activity_vertical_margin"
+        android:background="#fdfdfd"
```
Comment (gsintrepid) [inline app/src/main/res/layout/category_row_item.xml]: Use the dimension variable

File: app/src/main/res/layout/category_row_item.xml [hunk 8]
```diff
@@ -1,35 +1,65 @@
 <?xml version="1.0" encoding="utf-8"?>
-<android.support.v7.widget.CardView
+<android.support.design.widget.CoordinatorLayout
     xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto"
-    android:id="@+id/scoring_category_card_view"
-    android:layout_width="@dimen/category_card_width"
+    xmlns:card_view="http://schemas.android.com/tools"
```
Comment (gsintrepid) [inline app/src/main/res/layout/category_row_item.xml]: This is not used

File: app/build.gradle [hunk 9]
```diff
@@ -22,7 +22,7 @@ android {
 
     defaultConfig {
         applicationId "io.intrepid.contest"
-        minSdkVersion 17
+        minSdkVersion 19
```
Comment (gsintrepid) [inline app/build.gradle]: Why did you change this version? What in this PR is not available in sdk 17?

File: app/src/main/res/layout/category_row_item.xml (hunk 6 above)
Comment (abassawo) [inline app/src/main/res/layout/category_row_item.xml]: so when i use fontFamily the text doesn't actually look bold. lets talk about this

File: app/build.gradle (hunk 9 above)
Comment (abassawo) [inline app/build.gradle]: I thought I would need it for cardview elevation, but I don't think I will. reverting!

File: app/src/main/res/layout/category_row_item.xml (hunk 6 above)
Comment (gsintrepid) [inline app/src/main/res/layout/category_row_item.xml]: As we talked, disregard this comment. the right one is `android:textAppearance`. I filed a bug to fix all fonts in the app: https://intrepid.atlassian.net/browse/CON-160

File: app/src/main/res/layout/category_row_item.xml [hunk 10]
```diff
@@ -1,35 +1,63 @@
...
     android:layout_height="@dimen/category_card_height"
-    app:cardCornerRadius="@dimen/cardview_default_radius"
-    app:cardElevation="@dimen/cardview_default_elevation"
     >
 
-    <LinearLayout
+    <ImageView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:background="@android:color/transparent"
```
Comment (gsintrepid) [inline app/src/main/res/layout/category_row_item.xml]: Is this background making any difference? What's the default?
[2]

File: app/src/main/java/io/intrepid/contest/screens/contestcreation/editcategoriestocontest/EditCategoriesPresenter.java [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+
+    EditCategoriesPresenter(@NonNull View view, @NonNull PresenterConfiguration configuration) {
+        super(view, configuration);
+    }
+
+    @Override
+    public void onNextClicked(String name, String description) {
+            Timber.d("Adding new category " + name + " " + description);
+            view.addCategory(new Category(name, description));
+        view.showCategoriesList();
```
Comment (kentonguan) [inline app/src/main/java/io/intrepid/contest/screens/contestcreation/editcategoriestocontest/EditCategoriesPresenter.java]: formatting

File: app/src/main/java/io/intrepid/contest/screens/contestcreation/editcategoriestocontest/EditCategoryActivity.java [hunk 12]
```diff
@@ -1,25 +1,33 @@
...
     protected Fragment createFragment(Intent intent) {
-        return new AddCategoriesFragment();
+        return new EditCategoriesFragment();
+    }
+
+    public static Intent makeEditCategoryIntent(Context context, Category category) {
+        return new Intent(context, EditCategoryActivity.class)
+                .putExtra(CATEGORY_NAME, category.getName())
+                .putExtra(CATEGORY_DESCRIPTION, category.getDescription())
+                ;
```
Comment (kentonguan) [inline app/src/main/java/io/intrepid/contest/screens/contestcreation/editcategoriestocontest/EditCategoryActivity.java]: formatting

File: app/src/test/java/io/intrepid/contest/screens/contestcreation/editcategoriestocontest/EditCategoriesPresenterTest.java [hunk 13]
```diff
@@ -15,16 +15,16 @@
...
-    private AddCategoriesContract.Presenter addCategoryPresenter;
+    EditCategoriesContract.View mockView;
+    private EditCategoriesContract.Presenter addCategoryPresenter;
 
     @Before
     public void setup() {
         MockitoAnnotations.initMocks(this);
-        addCategoryPresenter = new AddCategoriesPresenter(mockView,
-                                                          TestPresenterConfiguration.createTestConfiguration());
+        addCategoryPresenter = new EditCategoriesPresenter(mockView,
```
Comment (kentonguan) [inline app/src/test/java/io/intrepid/contest/screens/contestcreation/editcategoriestocontest/EditCategoriesPresenterTest.java]: rename to editCategoryPresenter?

Comment (abassawo) [inline app/src/test/java/io/intrepid/contest/screens/contestcreation/editcategoriestocontest/EditCategoriesPresenterTest.java]: definitely
